subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuildRegistry.java

```diff
@@ -83,6 +84,11 @@ public boolean isSatisfiedBy(IncludedBuild includedBuild) {
         });
     }
 
+    @Override
+    public Collection<Pair<ModuleVersionIdentifier, ProjectComponentIdentifier>> getModuleToProjectMapping(IncludedBuild includedBuild) {
+        return ((IncludedBuildInternal)includedBuild).getAvailableModules();
```

Comment:
`IncludedBuildInternal` used to be in `dependencyManagement` because it needed some internal types from that project, so we couldn't move the type into `coreApi`.  I think we can move it now, although I sort of like that `coreApi` has less in it.  We can't add `compositeBuilds` as a dependency to `dependencyManagement` due to a cycle.

Reply:
👍  It's not worth changing.